### DISCUSSION TITLE
question about podman cgroup init process

### DISCUSSION THREAD

**Discussion by calvin2021y** [containers/podman#16564] [Q&A]
I try to start a rootless podman container by direct start by crun.

When I use podman inside podman (non-root user) with privileged option, I can move pid 1 into a new create cgroup.

```sh
podman --privileged -it --rm --systemd=false  mgoltzsche/podman:latest ash
cat /sys/fs/cgroup/cgroup.procs 
1
2
mkdir /sys/fs/cgroup/init
echo 1 > /sys/fs/cgroup/init/cgroup.procs 
cat /sys/fs/cgroup/init/cgroup.procs 
```

when I try start a container by crun with config.json, and from a cgroup like this:

```sh
cat /proc/self/cgroup
0::/user.slice/user-1001.slice/user@1001.service/user.slice/test.scope
crun --rootless=/tmp/test run test
# now int container shell
mkdir /sys/fs/cgroup/init
echo 1 > /sys/fs/cgroup/init/cgroup.procs 
cat /sys/fs/cgroup/init/cgroup.procs 
echo: write error: Not supported
cat /sys/fs/cgroup/cgroup.procs 
0
1
2
```

I am not sure why my crun container ` /sys/fs/cgroup/cgroup.procs` has a pid 0,  how can I avoid this by adust crun argument or `config.json`?

podman need move pid 1 into init to start work in rootless mode, `echo: write error: Not supported` error block this step.

**Comment by calvin2021y** [comment]
after use runc replace crun inside container, and other adjust the problem is fixed. 

not sure why crun not work.

**Comment by rhatdan** [comment]
@giuseppe @flouthoc PTAL
